src/libsync/propagateremotemkdir.cpp

```diff
@@ -44,7 +45,7 @@ PropagateRemoteMkdir::PropagateRemoteMkdir(OwncloudPropagator *propagator, const
         return;
     }
 
-    if (hasEncryptedAncestor()) {
+    if (hasEncryptedAncestor() || _isEncryptedRootFolder) {
```

Comment:
So I guess the condition really is `_item->_isEncrypted || hasEncryptedAncestor()`?